test/optimistic-erc20.spec.ts

```diff
@@ -0,0 +1,227 @@
+/* External Imports */
+import { ethers } from 'hardhat'
+import { Contract, Signer, ContractFactory } from 'ethers'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+/* Internal Imports */
+import { expect } from './setup'
+import deploymentsInfoL1 from '../deployments/l1/ERC20.json'
+import deploymentsInfoL2 from '../deployments/l2/ERC20.json'
+
+describe('ERC20', () => {
+  // Contracts
+  let ERC20_L1: any,
+    ERC20_L2: any,
+    privateKey: string,
+    // Providers
+    providerL1: JsonRpcProvider,
+    providerL2: JsonRpcProvider,
+    // Signers
+    walletL1_2: Signer,
+    walletL1_3: Signer,
+    walletL2_2: Signer,
+    walletL2_3: Signer,
+    // Deployers
+    deployerL1: Signer,
+    deployerL2: Signer
+
+  0.14429571
+  const name = 'Some Really Cool Token Name'
+  const initialSupplyL1 = 10000000 // For L1 contract
+  const initialSupplyL2 = 10000000 // For L2 contract
+
+  before('set up providers and wallets', () => {
+    privateKey = ethers.Wallet.createRandom().privateKey
+
+    providerL1 = new ethers.providers.JsonRpcProvider(process.env.L1_WEB3_URL)
+    providerL2 = new ethers.providers.JsonRpcProvider(process.env.L2_WEB3_URL)
```

Comment:
```suggestion
    provider= new ethers.providers.JsonRpcProvider((isL2 ? process.env.L2_WEB3_URL : process.env.L1_WEB3_URL))
```

Reply:
At a high level -refactor integration tests to be able to run against an L2 deployment or an L1 deployment based on a simple flag

Reply:
The devex we want is to run compilation + deployment + integration tests against a vanilla evm chain. Then just run a different set of scripts with a flag on (e.g. TARGET=ovm) to run the compilation + deployment + integration tests against a local OVM chain